source/core/index.ts

```diff
@@ -116,7 +117,13 @@ type CacheableRequestFn = (
 	cb?: (response: ServerResponse | ResponseLike) => void
 ) => CacheableRequest.Emitter;
 
-export interface Options extends URLOptions, SecureContextOptions {
+type CheckServerIdentityFn = (hostname: string, certificate: DetailedPeerCertificate) => Error | undefined | void;
```

Comment:
```suggestion
```

Reply:
The problem is that `DefinitelyTyped ` describe the `certificate` argument as `PeerCertificate`, but it's a `DetailedPeerCertificate` (the same thing, but 1 more property)